test/framework/turtles.go

```diff
@@ -94,3 +98,17 @@ func WaitForDeploymentsRemoved(ctx context.Context, input WaitForDeploymentsRemo
 		return false
 	}, intervals...).Should(BeTrue(), func() string { return capiframework.DescribeFailedDeployment(input, deployment) })
 }
+
+type VerifyCustomResourceHasBeenRemovedInput struct {
+	Config               *rest.Config
+	GroupVersionResource schema.GroupVersionResource
+}
+
+func VerifyCustomResourceHasBeenRemoved(ctx context.Context, input VerifyCustomResourceHasBeenRemovedInput) {
+	Byf("Verifying that custom resource %q has been removed", input.GroupVersionResource.String())
+	dynamicClient, err := dynamic.NewForConfig(input.Config)
+	Expect(err).NotTo(HaveOccurred(), "Failed to create dynamic client for config: %v", input.Config)
+	list, err := dynamicClient.Resource(input.GroupVersionResource).List(ctx, metav1.ListOptions{})
+	Expect(err).NotTo(HaveOccurred(), "Failed to list custom resource %q: %v", input.GroupVersionResource.String(), err)
+	Expect(list.Items).To(BeEmpty(), "Custom resource %q should have been removed, but found %d items", input.GroupVersionResource.String(), len(list.Items))
+}
```

Comment:
```suggestion
func VerifyCustomResourceHasBeenRemoved(ctx context.Context, input VerifyCustomResourceHasBeenRemovedInput) {
	Byf("Verifying that custom resource %q has been removed", input.GroupVersionKind.String())
	list := &unstructured.UnstructuredList{}
	list.SetGroupVersionKind(input.GroupVersionKind)
	err := input.Client.List(ctx, list, []client.ListOption{}...)
	Expect(err).NotTo(HaveOccurred(), "Failed to list custom resource %q: %v", input.GroupVersionKind.String(), err)
	Expect(list.Items).To(BeEmpty(), "Custom resource %q should have been removed, but found %d items", input.GroupVersionKind.String(), len(list.Items))
}
```

Reply:
can we use unstructured and a typed controller runtime client for consistency with other parts of the code?

Reply:
sure